opencga-catalog/src/main/java/org/opencb/opencga/catalog/monitor/executors/ExecutorFactory.java

```diff
@@ -21,29 +21,27 @@
 /**
  * Created by pfurio on 22/08/16.
  */
-public class ExecutorManager {
+public class ExecutorFactory {
 
     // TODO: Change for a map
-    private AbstractExecutor executor;
+    private BatchExecutor executor;
+
+    public ExecutorFactory(Configuration configuration) {
+        this.executor = new LocalExecutor();
 
-    public ExecutorManager(Configuration configuration) {
         if (configuration != null) {
             if (configuration.getExecution().getMode().equalsIgnoreCase("local")) {
```

Comment:
What about a switch/case statement?

Reply:
I'd throw an exception if the "mode" is not recognized, and only if it's null or empty, use a LocalExecutor.
It changes the behaviour, but it will remove a silent error.